NitroxServer/Communication/Packets/Processors/EntityMetadataUpdateProcessor.cs

```diff
@@ -45,5 +49,22 @@ private void SendUpdateToVisiblePlayers(EntityMetadataUpdate packet, Player send
                 }
             }
         }
+
+        // TODO: replace temporary code with a serious implementation
+        private void ProcessMetadata(Entity entity, EntityMetadata metadata)
+        {
+            if (metadata is PlayerMetadata playerMetadata)
+            {
+                Player player = playerManager.GetConnectedPlayers().Where(p => p.GameObjectId == entity.Id).FirstOrDefault();
+                if (player != null)
+                {
+                    player.EquippedItems.Clear();
+                    foreach (PlayerMetadata.EquippedItem item in playerMetadata.EquippedItems)
+                    {
+                        player.EquippedItems.Add(item.Slot, item.Id);
+                    }
+                }
+            }
+        }
```

Comment:
```suggestion
    private bool ProcessMetadata(Player sendingPlayer, Entity entity, EntityMetadata metadata)
    {
        return metadata switch
        {
            PlayerMetadata playerMetadata => ProcessPlayerMetadata(sendingPlayer, entity, playerMetadata),
            // By default, entities have no security layer
            _ => true,
        };
    }

    private static bool ProcessPlayerMetadata(Player sendingPlayer, Entity entity, PlayerMetadata playerMetadata)
    {
        // Security layer to ensure a player can only modify its own entity
        if (sendingPlayer.GameObjectId == entity.Id)
        {
            sendingPlayer.EquippedItems.Clear();
            foreach (PlayerMetadata.EquippedItem item in playerMetadata.EquippedItems)
            {
                sendingPlayer.EquippedItems.Add(item.Slot, item.Id);
            }
            return true;
        }
        
        Log.WarnOnce($"Player: {sendingPlayer.Name} tried updating metadata of another player's entity {entity.Id}");
        return false;
    }
```
A switch for ``ProcessMetadata`` will be required for the efficiency this method needs